core/modules/qdisp/Executive.cc

```diff
@@ -223,74 +233,81 @@ bool Executive::join() {
     }
-
-    LOGF_INFO("Query exec finish. %1% dispatched." % _requestCount);
-    _reportStatuses();
-    if(sCount != _requestCount) {
-        LOGF_INFO("Query exec:. %1% != %2%" % _requestCount % sCount);
+    if(sCount == _requestCount) {
+        LOGF_INFO("Query execution succeeded: %1% jobs dispatched and completed." % _requestCount);
+    } else {
+        LOGF_ERROR("Query execution failed: %1% jobs dispatched, but only %2% jobs completed" % _requestCount % sCount);
     }
+    _updateProxyMessages();
     bool empty = (sCount == _requestCount);
     _empty.set(empty);
     LOGF_DEBUG("Flag set to _empty=%1% sCount=%2% requestCount=%3%" % empty % sCount % _requestCount);
     return empty;
 }
 
-void Executive::markCompleted(int refNum, bool success) {
-    ResponseRequester::Error e;
-    LOGF_INFO("Executive::markCompleted(%1%,%2%)" % refNum % success);
+void Executive::markCompleted(int jobId, bool success) {
+    ResponseRequester::Error err;
+    LOGF(getLogger(), LOG_LVL_INFO, "Executive::markCompleted(%1%,%2%)" % jobId % success);
     if(!success) {
         {
             std::lock_guard<std::mutex> lock(_requestersMutex);
-            RequesterMap::iterator i = _requesters.find(refNum);
+            RequesterMap::iterator i = _requesters.find(jobId);
             if(i != _requesters.end()) {
-                e = i->second->getError();
+                err = i->second->getError();
             } else {
                 std::string msg =
                     (boost::format("Executive::markCompleted(%1%) "
                                    "failed to find tracked id=%2% size=%3%")
-                     % (void*)this % refNum % _requesters.size()).str();
+                     % (void*)this % jobId % _requesters.size()).str();
                 throw Bug(msg);
             }
         }
+        LOGF(getLogger(), LOG_LVL_ERROR,
+             "Executive: error executing jobId=%1%: %2% (status: %3%)" % jobId % err % err.getStatus());
         {
             std::lock_guard<std::mutex> lock(_statusesMutex);
-            _statuses[refNum]->report(ExecStatus::RESULT_ERROR, 1);
+            _statuses[jobId]->updateInfo(JobStatus::RESULT_ERROR, err.getCode(), err.getMsg());
+        }
+        {
+            std::lock_guard<std::mutex> lock(_errorsMutex);
+            _multiError.push_back(err);
+            LOGF(getLogger(), LOG_LVL_TRACE, "Currently %2% registered errors: %1%" % _multiError % _multiError.size());
         }
-        LOGF_ERROR("Executive: error executing refnum=%1%. Code=%2% %3%" %
-                   refNum % e.code % e.msg);
     }
-    _unTrack(refNum);
+    _unTrack(jobId);
     if(!success) {
-        LOGF_ERROR("Executive: requesting squash (cause refnum=%1% with code=%2% %3%)" % refNum % e.code % e.msg);
+        LOGF(getLogger(), LOG_LVL_ERROR,
+                 "Executive: requesting squash, cause: jobId=%1% failed (code=%2% %3%)" % jobId % err.getCode() % err.getMsg());
```

Comment:
Indentation, and make line shorter, it's hard to review code which is wider than github panel size.